server/src/main/java/org/spine3/server/storage/memory/InMemoryAggregateStorage.java

```diff
@@ -60,62 +62,54 @@ protected InMemoryAggregateStorage(boolean multitenant) {
 
     @Override
     protected void writeRecord(I id, AggregateStorageRecord record) {
-        recordMap.put(id, record);
+        getStorage().put(id, record);
     }
 
     @Override
     protected Iterator<AggregateStorageRecord> historyBackward(I id) {
         checkNotNull(id);
-        final Collection<AggregateStorageRecord> records = recordMap.get(id);
+        final List<AggregateStorageRecord> records = getStorage().getHistoryBackward(id);
         return records.iterator();
     }
 
     @Override
     protected int readEventCountAfterLastSnapshot(I id) {
         checkNotClosed();
-        final Integer count = eventCountMap.get(id);
-        if (count == null) {
-            return 0;
-        }
-        return count;
+        final int result = getStorage().getEventCount(id);
+        return result;
     }
 
     @Override
     protected void writeEventCountAfterLastSnapshot(I id, int eventCount) {
         checkNotClosed();
-        eventCountMap.put(id, eventCount);
+        getStorage().putEventCount(id, eventCount);
     }
 
-    /** Used for sorting by timestamp descending (from newer to older). */
-    private static class AggregateStorageRecordReverseComparator implements Comparator<AggregateStorageRecord>,
-                                                                            Serializable {
-        private static final long serialVersionUID = 0L;
+    @Override
+    protected boolean markArchived(I id) {
+        final AggregateStatus currentStatus = getStorage().getStatus(id);
```

Comment:
That could lead to the performance issue.

If the aggregate is NOT archived, we'll be calling the storage twice (which is an expensive operation). And even if it is archived, we'll still call the storage at least once.

I would leave a comment somewhere in the JavaDocs that for a real storage (not `InMemory...` one) they should probably always update the state.